code/src/pages/MovieDetail.js

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import "./MovieDetail.css";
+
+export const MovieDetail = () => {
+  const { movieId } = useParams();
+  const [movie, setMovie] = useState([]);
+
+  useEffect(() => {
+    fetch(
+      `https://api.themoviedb.org/3/movie/${movieId}?api_key=8a43e19df67e81dfd10b5f88d05f471f&language=en-US`
+    )
+      .then(res => res.json())
+      .then(json => {
+        setMovie(json);
+      });
+  }, [movieId]);
+
+  return (
+    <section className="detail-page">
+      <div
+        className="background"
+        style={{
+          backgroundImage: `linear-gradient(180deg, rgba(0,0,0,0.5) 70%, rgba(0,0,0,1) 100%), url(https://image.tmdb.org/t/p/w1280${movie.backdrop_path})`
```

Comment:
I think the error messages you get has to do with that some of the movies doesn't have a backdrop_path or a poster-path or similar. So you get a undefined there instead of an actual path. Or that it happens too quickly before the fetch is done.
To fix this you could work with some defensive programming, which means that we're planning for what could go wring, just to be safe. So you could make sure that fetch is done and the movie variable is actually "filled" with some data by doing something like:
`if (movie) return <section className="background"> ... `
Or the other way around, to add some kind of Loader to be shown until the fetch is done.
`if (!movie) return <LoadingSpinner />`